app/src/main/java/espertolabs/esperto_ble_watch/SummaryActivity.java

```diff
@@ -650,8 +652,13 @@ public void run() {
                 //send request via apigateway
                 String response = handler.getHeartRate(userId);
                 Gson g = new Gson();
-                HeartRate hr = g.fromJson(response, HeartRate.class);
-                userHR.setCurrentHR(hr.getCurrentHR());
+                if(response != ""){
+                    HeartRate hr = g.fromJson(response, HeartRate.class);
+                    userHR.setCurrentHR(hr.getCurrentHR());
```

Comment:
Does HR have the same GSON issue as steps?

Reply:
Nah, it's not an issue with HR. 
The issue with steps probably due to either bad/a lack of data coming in from the request, or the fact that the StepCount class doesn't reflect the data coming in from the request.